common.eai

```diff
@@ -6060,28 +6089,27 @@ endfunction
 
 //============================================================================
 function GetExpFoe takes nothing returns unit
-  local group g = null
-  local group t = null
+  local group g = CopyGroup(current_expansion_creeps)
   local unit u = null
   local location l = GetUnitLoc(current_expansion)
+  local real i = expansion_radius
   //call Trace("Copy group")
-  
-  set g = CopyGroup(current_expansion_creeps)
   set g = SelectByAlive(g, true)
   set u = GetNearestOfGroup(g, l)
   if l != null then
 	call RemoveLocation(l)
 	set l = null
   endif
-  if u == null then
-	set t = CreateGroup()
-    call GroupEnumUnitsInRange(t, GetUnitX(current_expansion), GetUnitY(current_expansion), expansion_radius, null)
-    set t = SelectByPlayer(t, Player(PLAYER_NEUTRAL_AGGRESSIVE), true)
-    set t = SelectByAlive(t, true) 	
-	set u = FirstOfGroup(t)
-	call DestroyGroup(t)
-    set t = null
-  endif
+  loop
```

Comment:
This loop can crash the AI script as it can become infinite if there are no units in the area. Adding a sleep won't help either as then the attack function while it doesn't crash, just halts in a loop that won't end.